spring-cloud-data-rest-client/src/main/java/org/springframework/cloud/data/rest/client/CloudDataTemplate.java

```diff
@@ -55,10 +60,16 @@ public CloudDataTemplate(URI baseURI) {
 		resources.put("streams/definitions", new UriTemplate(resourceSupport.getLink("streams").getHref() + "/definitions"));
 		resources.put("streams/deployments", new UriTemplate(resourceSupport.getLink("streams").getHref() + "/deployments"));
 		this.streamTemplate = new StreamTemplate(restTemplate, resources);
+		this.counterTemplate = new CounterTemplate(restTemplate, resourceSupport);
```

Comment:
We need to decide which approach is best here so we can be consistent. I like that the StreamTemplate is currently handed only what it needs and that it's simply a Map or UriTemplates at that point.